test/ng/directive/asSpec.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+describe('as', function() {
+
+  it('should bind in the current scope the controller of a component', function() {
+    var expected = {some: 'value'};
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        this.property = expected;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      var $ctrl = {undamaged: true};
+      scope.$ctrl = $ctrl;
+      dealoc($compile('<my-component as="$ctrl.myComponent"></my-component>')(scope));
+
+      expect(scope.$ctrl).toBe($ctrl);
+      expect(scope.$ctrl.undamaged).toBe(true);
+      expect($ctrl.myComponent).not.toBeUndefined();
+      expect($ctrl.myComponent && $ctrl.myComponent.property).toBe(expected);
+      scope.$destroy();
```

Comment:
After some investigation it seems that `testabilityPatch` applies an `afterEach` to all tests, that automatically cleans up `$rootElement` and `$rootScope`. This means that you don't have to manually call `dealoc` if at least one of the following conditions holds:

1. Any elements you create are in the DOM tree of the `$rootElement` (e.g. you call `$rootElement.append(yourCreatedElement)`).
2. No other scope than `$rootScope` is sued for linking.

E.g. in this test, if you keep using `$rootScope.$new()`, then you need to `dealoc` (either the element or the scope) - yup, scopes can be `dealoc`'d too :smiley:
If you switch to using `$rootScope`, then I think you don't need `dealoc`ing.

The good thing is that if you don't clean up properly, you'll get some warnings in the console, so you know something's wrong.